docker-compose.yml

```diff
@@ -0,0 +1,210 @@
+    depends_on:
+      angular-build:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8181/ || exit 1
+      timeout: 10s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    entrypoint: /google-cloud-sdk/bin/dev_appserver.py
+    command: 
+      - "/app/oppia/app_dev_docker.yaml"
+      - "--runtime"
+      - "python38"
+      - "--host=0.0.0.0"
+      - "--port=8181"
+      - "--admin_host=0.0.0.0"
+      - "--admin_port=8000"
+      - "--skip_sdk_update_check=True"
+
+  datastore:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-cloud-datastore
+    ports:
+      - "8089:8089"
+    environment:
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+    depends_on:
+      angular-build:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8089/ || exit 1
+      timeout: 10s
+      interval: 2m
+      retries: 5
+      start_period: 20m
+    entrypoint: /google-cloud-sdk/bin/gcloud
+    command:
+      - "beta"
+      - "emulators"
+      - "datastore"
+      - "start"
+      - "--project"
+      - "dev-project-id"
+      - "--data-dir"
+      - "/app/cloud_datastore_emulator_cache"
+      - "--host-port=0.0.0.0:8089"
+      - "--consistency=1.0"
+      - "--quiet"
+      - "--no-store-on-disk"
+
+  firebase:
+    image: node@sha256:48650ae5c5dc9fbd0d0559fd1248e779f7214ba99434420a089bc7853aeefb65
+    container_name: oppia-firebase-emulator
+    ports:
+      - "9099:9099"
+      - "4000:4000"
+    volumes:
+      - ./.firebase.json:/app/oppia/.firebase.json
+      - shared_node_modules:/app/oppia/node_modules
+    environment:
+      - oppia_is_dockerized=true
+    healthcheck:
+      test: wget --spider http://localhost:4000/auth/ && echo 'y' | /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js emulators:export /app/firebase_emulator_cache --project=dev-project-id || exit 1
+      timeout: 20s
+      interval: 20s
+      retries: 5
+    entrypoint:
+      - /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js
+      - "emulators:start"
+      - "--only"
+      - "auth"
+      - "--project=dev-project-id"
+      - "--config=/app/oppia/.firebase.json"
+    command:
+      - "--import=/app/firebase_emulator_cache"
+      - "--export-on-exit=/app/firebase_emulator_cache"
+
+  redis:
+    image: redis@sha256:aebc052b5a7f50d79f00a4deee22109e2ecf22f439140a03f2c9770055ffec7b
+    container_name: oppia-redis
+    ports:
+      - "6379:6379"
+    environment:
+      - PATH=/usr/local/bin/
+    volumes:
+      - oppia_redis_dump:/app/oppia/dump.db
+      - ./redis_docker.conf:/app/oppia/redis_docker.conf
+    healthcheck:
+      test: redis-cli ping
+      timeout: 5s
+      interval: 30s
+      retries: 5
+    command: redis-server /app/oppia/redis_docker.conf
```

Comment:
Where are we running this command and where will it form the "dump.db"? Will it be inside app/oppia/?

Reply:
> Where are we running this command 
didn't get properly, the command will start the redis server

> where will it form the "dump.db"
yes, it is inside `/app/oppia`

Reply:
> didn't get properly, the command will start the redis server

what's the working dir. from where you are running the `redis-server` command and what will be the path of the dump.db? (You can confirm it by starting the server and open the terminal in the container and look for the dump.db file)

Reply:
okay, so we are using redis dockerhub image, so thats why no need to specify the path for `redis-server`.
I confirm that dumb.dp is created within `/app/oppia`

Reply:
What's the workingdir (cwd) where we are running redis-server. For example, if we are running the `redis-server` command from `/usr/` so will it generate `dump.db` inside `/usr/`, if yes then knowing working dir is important here to cache the dump.db correctly in named volumes.

Reply:
oh no, please check the redis_docker.conf file. it specifies every required configs, so dumb.db is created within /app/oppia

Reply:
I see `dir ./` which again depends on `.` i.e, pwd. Can you please check the pwd of the container when running redis-server command and attach a screenshot?

Reply:
I had to rerun the server to test it as I'm worried we are saving it inside /data (After reading Dockerfile of redis [line 11](https://hub.docker.com/layers/library/redis/latest/images/sha256-4e7bd0aec37b626ecd11c37892dc86e142b3e1bbe2694ee524dcb518ece6b7ff?context=explore)). I know we can go wrong sometimes but I would highly suggest (please) validating things practically.

![image](https://github.com/oppia/oppia/assets/16653571/78e576f0-1286-4302-9ddf-d5407217db87)

Reply:
I would suggest keeping it inside /data i.e, `/data/dump.db` as it's the default path and sounds like a better path for dumped data. wdyt?

Reply:
changed the volume mount, please check if it is fine!